src/components/Create/components/Allocation/AddEditAllocationModal.tsx

```diff
@@ -0,0 +1,342 @@
+      setAmountType(hasSpecificFundingTarget ? 'amount' : 'percentage')
+      return
+    }
+
+    setRecipient(
+      editingData.projectId !== undefined ? 'juiceboxProject' : 'walletAddress',
+    )
+    setAmountType(hasSpecificFundingTarget ? 'amount' : 'percentage')
+    form.setFieldsValue({
+      juiceboxProjectId: editingData.projectId,
+      address: editingData.beneficiary,
+      amount: {
+        percentage: editingData.percent.toString(),
+        amount: totalAllocationAmount
+          ? amountFromPercent({
+              percent: editingData.percent,
+              amount: totalAllocationAmount.toString(),
+            }).toString()
+          : undefined,
+      },
+      lockedUntil: editingData.lockedUntil
+        ? Moment.default(editingData.lockedUntil * 1000)
+        : undefined,
+    })
+  }, [
+    hasSpecificFundingTarget,
+    editingData,
+    form,
+    visible,
+    totalAllocationAmount,
+  ])
+
+  const onModalOk = useCallback(async () => {
+    const fields = await form.validateFields()
+    const result: AddEditAllocationModalResult = {
+      id: `${fields.address}${
+        fields.juiceboxProjectId ? `-${fields.juiceboxProjectId}` : ''
+      }`,
+      beneficiary: fields.address,
+      projectId: fields.juiceboxProjectId,
+      percent: parseFloat(fields.amount!.percentage),
+      lockedUntil: fields.lockedUntil
+        ? Math.round(fields.lockedUntil.valueOf() / 1000)
+        : undefined,
+      preferClaimed: undefined,
+      allocator: undefined,
+    }
+    onOk(result)
+    form.resetFields()
+  }, [form, onOk])
+
+  const onModalCancel = useCallback(() => {
+    onCancel()
+    form.resetFields()
+  }, [form, onCancel])
+
+  const addressLabel =
+    recipient === 'juiceboxProject' ? t`Token beneficiary address` : t`Address`
+
+  return (
+    <Modal
+      title={<h2>{isEditing ? t`Edit payout` : t`Add new payout`}</h2>}
+      okText={t`Add payout`}
+      visible={visible}
+      onOk={onModalOk}
+      onCancel={onModalCancel}
+      destroyOnClose
+    >
+      <Form form={form} preserve={false} colon={false} layout="vertical">
+        {isEditing && hasSpecificFundingTarget && (
+          <Radio.Group
+            optionType="button"
+            defaultValue="amount"
+            value={amountType}
+            onChange={e => setAmountType(e.target.value)}
+          >
+            <Radio value="amount">
+              <Trans>Amounts</Trans>
+            </Radio>
+            <Radio value="percentage">
+              <Trans>Percentages</Trans>
+            </Radio>
+          </Radio.Group>
+        )}
+        <Form.Item label={t`Recipient`}>
+          <Radio.Group
+            value={recipient}
+            onChange={e => setRecipient(e.target.value)}
+          >
+            <Radio value="walletAddress">
+              <Trans>Wallet Address</Trans>
+            </Radio>
+            <Radio value="juiceboxProject">
+              <Trans>Juicebox Project</Trans>
```

Comment:
**nitpick** IMO its cool to not capitalize each word unless theres a specific reason to

Reply:
Happy to role with it, but since i am mostly following strath design, lets leave it to the end